dotnet/InlineProgram/Program.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Pulumi.Automation;
+
+namespace InlineProgram
```

Comment:
The inline program is failing with the following exception once it reaches the `pulumi up` step:

```
PS C:\git\pulumi-auto-examples\dotnet\InlineProgram> dotnet run                                                                                                                                                                                                                                                                                        successfully initialized stack
installing plugins...
plugins installed
setting up config...
config set
refreshing stack...
refresh complete
updating stack...
info: Microsoft.Hosting.Lifetime[0]
      Now listening on: http://0.0.0.0:58710
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\git\pulumi-auto-examples\dotnet\InlineProgram
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 POST http://127.0.0.1:58710/pulumirpc.LanguageRuntime/GetRequiredPlugins application/grpc
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
      Executing endpoint 'gRPC - /pulumirpc.LanguageRuntime/GetRequiredPlugins'
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint 'gRPC - /pulumirpc.LanguageRuntime/GetRequiredPlugins'
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 POST http://127.0.0.1:58710/pulumirpc.LanguageRuntime/Run application/grpc
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 71.4509ms 200 application/grpc
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
      Executing endpoint 'gRPC - /pulumirpc.LanguageRuntime/Run'
info: Microsoft.Hosting.Lifetime[0]
      Application is shutting down...
Unhandled exception. fail: Grpc.AspNetCore.Server.ServerCallHandler[6]
      Error when executing service method 'Run'.
Pulumi.LogException: Error occurred during logging
 ---> Grpc.Core.RpcException: Status(StatusCode="Unavailable", Detail="failed to connect to all addresses", DebugException="Grpc.Core.Internal.CoreErrorDetailException: {"created":"@1614445151.998000000","description":"Failed to pick subchannel","file":"..\..\..\src\core\ext\filters\client_channel\client_channel.cc","file_line":4143,"referenced_errors":[{"created":"@1614445151.989000000","description":"failed to connect to all addresses","file":"..\..\..\src\core\ext\filters\client_channel\lb_policy\pick_first\pick_first.cc","file_line":398,"grpc_status":14}]}")
   at Pulumi.GrpcEngine.LogAsync(LogRequest request)
   at Pulumi.Deployment.Logger.LogAsync(LogSeverity severity, String message, Resource resource, Nullable`1 streamId, Nullable`1 ephemeral)
   --- End of inner exception stack trace ---
   at Pulumi.Deployment.Logger.LogAsync(LogSeverity severity, String message, Resource resource, Nullable`1 streamId, Nullable`1 ephemeral)
   at Pulumi.Deployment.Runner.HandleExceptionAsync(Exception exception)
   at Pulumi.Deployment.Runner.WhileRunningAsync()
   at Pulumi.Automation.PulumiFnInline.InvokeAsync(IRunner runner, CancellationToken cancellationToken)
   at Pulumi.Automation.LanguageRuntimeService.Run(RunRequest request, ServerCallContext context)
   at Grpc.Shared.Server.UnaryServerMethodInvoker`3.AwaitInvoker(Task`1 invokerTask, GrpcActivatorHandle`1 serviceHandle)
   at Grpc.Shared.Server.UnaryServerMethodInvoker`3.AwaitInvoker(Task`1 invokerTask, GrpcActivatorHandle`1 serviceHandle)
   at Grpc.AspNetCore.Server.Internal.CallHandlers.UnaryServerCallHandler`3.HandleCallAsyncCore(HttpContext httpContext, HttpContextServerCallContext serverCallContext)
   at Grpc.AspNetCore.Server.Internal.CallHandlers.ServerCallHandlerBase`3.<HandleCallAsync>g__AwaitHandleCall|8_0(HttpContextServerCallContext serverCallContext, Method`2 method, Task handleCall)
System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at Microsoft.Extensions.Hosting.Internal.Host.StopAsync(CancellationToken cancellationToken)
   at Pulumi.Automation.WorkspaceStack.InlineLanguageHost.DisposeAsync()
   at Pulumi.Automation.WorkspaceStack.UpAsync(UpOptions options, CancellationToken cancellationToken)
   at InlineProgram.Program.Main(String[] args) in C:\git\pulumi-auto-examples\dotnet\InlineProgram\Program.cs:line 115
   at InlineProgram.Program.<Main>(String[] args)
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint 'gRPC - /pulumirpc.LanguageRuntime/Run'
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 78942.6078ms 0 application/grpc
```

Which is... odd. Not sure what's going on here. Also maybe we want to revisit default logging from the internal language host we spin up.

Reply:
This appears to have resolved itself not sure what the issue was. Must've been some endpoint that was unreachable at that moment.